src/main/java/echoserver/HostServer.java

```diff
@@ -0,0 +1,34 @@
+package echoserver;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+
+public class HostServer{
+    Integer port;
+
+    public HostServer(ServerSocket socket) {
+        this.socket = socket;
+        this.port = socket.getLocalPort();
+    }
+
+    public ClientConnection listenForConnection() throws IOException {
+        Socket connectedClient = this.socket.accept();
+
+        return new ClientConnection(connectedClient);
```

Comment:
😄 Nice! Definitely an exception to our rule that we cannot new up a class inside another class. :tada: